database/migrations/2020_08_18_032818_add_member_character_slug.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddMemberCharacterSlug extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('characters', function (Blueprint $table) {
+            $table->string('slug')->nullable()->after('name');
+        });
+
+        Schema::table('members', function (Blueprint $table) {
+            $table->string('slug')->nullable()->after('username');
```

Comment:
We're going to use this slug to lookup members from the database, so I think we should index both of these new slugs.

The slug function in PHP also limits the slug's length to 50 characters, and that's loads for a username. I think we can also set these fields to be 50 characters for faster indexing+lookup (I am sort of assuming it's faster).

These fields are also required to load a page for either member or character, so we shouldn't allow null on them. (`username` and `name` also have NOT NULL set to true)

```
$table->string('slug', 50)->after('username')->index();
```

Reply:
Guilds and instances have their slugs set to 255 characters, so I think there's an opportunity to optimize those fields too at some point.

Reply:
updated as per your feedback.